linera-chain/src/data_types.rs

```diff
@@ -21,7 +21,10 @@ use linera_execution::{
 };
 use serde::{de::Deserializer, Deserialize, Serialize};
 
-use crate::{types::ValidatedBlockCertificate, ChainError};
+use crate::{
+    types::{Hashed, ValidatedBlockCertificate},
```

Comment:
I just noticed that this is hard to find: You have to go to `lib.rs` to see that the `types` module re-exports everything from `block` and `certificate`, and then guess that `Hashed` is defined in the latter.

Also, it's weird that we have a `types` and a `data_types` submodule.

Reply:
I disagree:
1. It has a unique name (so if you do full-text search you'll find it, unless with many other things in our codebase)
2. All IDEs can easily find the symbol (even GitHub)
3. Re-exporting is a common approach. If you prefer I can leave the fully qualified module path (`certificate::ValidatedBlockCertificate`) but IMHO exporting all the "important bits" under the same path (while keeping their code separate for readability) is a better dev experience.

Reply:
OK, but having `types` _and_ `data_types` is still weird.

Reply:
`data_types` will go away in the future. Or at least its content will change. The end-game is that the `Certificate` type will be removed and replaced with `types::Certificate` and rest of the code will be sliced up. `data_types.rs` has almost 1500 lines of code - it's very unreadable.